dtdl-tools/src/dtdl-validator/DtdlValidator.cs

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+// SPDX-License-Identifier: MIT
+
+using DTDLParser;
+using DTDLParser.Models;
+using System;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.NamingConventionBinder;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+class Program
+{
+    // Exit codes.
+    private const int EXIT_SUCCESS = 0;
+    private const int EXIT_FAILURE = 1;
+
+    /// <summary>
+    /// This method validates all of the DTDL files with the provided extension that are located
+    /// under the provided directory.
+    /// </summary>
+    /// <param name="directory">The directory that contains the DTDL files that we wish to validate.</param>
+    /// <param name="extension">The extension used by the DTDL files.</param>
+    /// <returns>
+    /// EXIT_SUCCESS when all if the DTDL files are valid.
+    /// EXIT_FAILURE when any of the DTDL files are NOT valid.
+    /// </returns>
+    static int ValidateDtdl(DirectoryInfo directory, String extension)
+    {
+        if (!directory.Exists)
+        {
+            Console.WriteLine($"Directory {directory.FullName} does not exist.");
+            return EXIT_FAILURE;
+        }
+
+        var files = Directory.GetFiles(directory.FullName, $"*.{extension}", SearchOption.AllDirectories);
+
+        if (!files.Any())
+        {
+            Console.WriteLine($"No files with extension .{extension} found in directory {directory}");
+            return EXIT_FAILURE;
+        }
+
+        var parser = new ModelParser();
+
+        bool failureOccured = false;
+        foreach (var file in files)
+        {
+            try
+            {
+                var dtdl = File.ReadAllText(file);
+                parser.ParseAsync(dtdl).GetAwaiter().GetResult();
```

Comment:
either document the reason that we need to run this synchronously or make `ValidateDtdl` async and use `await` here

Reply:
It is being called from within a handler.

Reply:
this doesn't change my recommendation. async code should be run in an async context when possible, and as far as I can tell it's not required to run this sync